### PR TITLE
BookStore 

### PR DESCRIPTION
**Pull request by AhmadiGu**
### Connect redux to react

- add provider
- use hooks useSelector, useDispitch 
-  able to add and remove functionalities

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    case GATEGORY_STATUS:
+      return {
+        categories: [],
+        status: 'Under construction',
+      };
+
+    default:
+      return state;
+  }
+};
```
Comment (sja-thedude) [inline src/redux/categories/categories.js:23]: - [ ] As per the requirements, kindly make sure to display "Under Construction" when the Check status button is clicked.

https://user-images.githubusercontent.com/70269507/210616018-e045b40d-a37e-44a3-8b01-c700199f27bd.mov

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @AhmadiGu, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters passing 
✔️ Correct gitflow
✔️ Professional readme

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -1,70 +1,167 @@
...
+
+
+👤 **Author1**
+
+- GitHub: [@githubhandle](https://github.com/AhmadiGu/)
+
+ 
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
```
Comment (danmainah) [inline README.md:127]: - [ ] [optional] It would have been nice if you added a future feature section in your readme file.

**Review by danmainah — APPROVED** [review]
Hi @AhmadiGu , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**